### PR TITLE
Adds walk support

### LINKED ISSUE #9: Help with implementing in ng-annotate-patched
I was hoping to integrate this plugin into [ng-annotate-patched](https://github.com/bluetech/ng-annotate-patched), so that dynamic imports could be used with webpack and ng-annotate. Sorry if this isn't an issue with this plugin, but even a pointer would be appreciated.

The issue seems to be that when this line is executed in the other project ([source code](https://github.com/bluetech/ng-annotate-patched/blob/fork/src/traverse.js)):

```js
const walk = require("acorn/dist/walk");
// ...
walk.base[override || node.type](node, st, c);
```
I get this error `Module build failed: TypeError: walk.base[(override || node.type)] is not a function` because `override || node.type` evaluates to `Import` which is missing in `walk.base`. I guess this is likely something added by this plugin and so looking in the original acorn module breaks?

So far the code I have changed in [this file](https://github.com/bluetech/ng-annotate-patched/blob/fork/src/ng-annotate-main.js) was instead of

```
const parser = require("acorn").parse;
```

I tried both

```
const parser = require('acorn-dynamic-import/lib/inject').default(require('acorn')).parse;
// const parser = require('acorn-dynamic-import').default.parse;
```

and I added `plugins: { dynamicImport: true },` into the options passed into the parser. 

Thanks for any suggestions!

Related Issue:

https://github.com/bluetech/ng-annotate-patched/issues/6

### PR DESCRIPTION
**Pull request by CodySchaaf**
Since the provided `Import` node doesn't have any meta data, I just copied the implementation of the other import nodes from the base implementation of acorn [here](https://github.com/ternjs/acorn/blob/191bd45a526b70c0beaf855f2773cb5c05969345/src/walk/index.js#L349), which essentially just ignores the node. 

This commit resolves #9

### CODE REVIEW

**Comment by CodySchaaf** [comment]
Linting is clean locally, seems like an npm failure on travis?

**Comment by CodySchaaf** [comment]
@kesne do you have any interest in merging this change?

**Comment by adrianheine** [comment]
I think if you retrigger travis this will pass. At least #11 passed, so it was probably an intermediate failure, since you don't touch that area at all.

File: package.json [hunk 1]
```diff
@@ -41,5 +41,5 @@
     "rimraf": "^2.5.4",
     "safe-publish-latest": "^1.1.1"
   },
-  "version": "2.0.2"
+  "version": "2.1.0"
```
Comment (adrianheine) [inline package.json]: I would not bump the version in a feature pull request.

File: src/walk.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import * as walk from 'acorn/dist/walk';
+import { DynamicImportKey } from './inject';
+
+export function inject(injectableWalk) {
+  return { ...injectableWalk,
+    base: {
+      ...injectableWalk.base,
+      [DynamicImportKey]: () => {},
+    } };
+}
```
Comment (adrianheine) [inline src/walk.js]: This will only work in node >= 8.

File: src/walk.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+export default inject(walk);
+
```
Comment (adrianheine) [inline src/walk.js]: Trailing whitespace

File: README.md [hunk 4]
```diff
@@ -28,6 +28,21 @@ const ast = acorn.parse(code, {
...
+```js
+import walk from 'acorn-dynamic-import/lib/walk';
+// or...
+const dynamicImportWalk = require('acorn-dynamic-import/lib/walk').default;
+```
+
+Or you can use the injectable version for injecting the new walk functionality into your own version of Acorn like this:
+
+```js
+const walk = require('acorn-dynamic-import/lib/walk').inject(require("acorn/dist/walk"));
```
Comment (adrianheine) [inline README.md]: I think it would look better if you would agree on one kind of quotation marks.

**Review by adrianheine — CHANGES_REQUESTED** [review]

File: src/walk.js (hunk 2 above)
Comment (CodySchaaf) [reply src/walk.js]: Since the node_module points to a built version I didn't think that would matter, but I can change to `Object.assign` for people that use this directly.

File: package.json (hunk 1 above)
Comment (CodySchaaf) [reply package.json]: Yeah I was trying to figure out how this was usually handled. Is the version usually updated in a separate pr to make the npm package available?

**Comment by CodySchaaf** [comment]
Still willing to get this merged into this project and get back to using this version, but for now I have forked it and am using https://www.npmjs.com/package/acorn-dynamic-import-with-walk-support

File: package.json [hunk 5]
```diff
@@ -1,13 +1,16 @@
 {
-  "name": "acorn-dynamic-import",
-  "description": "Support dynamic imports in acorn",
+  "name": "acorn-dynamic-import-with-walk-support",
```
Comment (kesne) [inline package.json]: Revert these changes.

File: README.md [hunk 6]
```diff
@@ -28,6 +28,21 @@ const ast = acorn.parse(code, {
...
+```js
+import walk from 'acorn-dynamic-import/lib/walk';
+// or...
+const dynamicImportWalk = require('acorn-dynamic-import/lib/walk').default;
+```
+
+Or you can use the injectable version for injecting the new walk functionality into your own version of Acorn like this:
+
+```js
+const walk = require('acorn-dynamic-import/lib/walk').inject(require('acorn/dist/walk'));
```
Comment (kesne) [inline README.md]: This probably should be this:

```js
import { inject } from 'acorn-dynamic-import/lib/walk';
import acornWalk from 'acorn/dist/walk';

const walk = inject(acornWalk);
```

File: src/walk.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import * as walk from 'acorn/dist/walk';
+import { DynamicImportKey } from './inject';
+
+export function inject(injectableWalk) {
+  return Object.assign({}, injectableWalk, {
+    base: Object.assign({}, injectableWalk.base, {
+      [DynamicImportKey]: () => {},
```
Comment (kesne) [inline src/walk.js]: Rather than using an arrow function here you could also just use object function shorthand:
```js
{
  [DynamicImportKey]() {},
}
```

**Comment by kesne** [comment]
@CodySchaaf Sorry I missed this, I'm bad at this OSS thing. I left a few comments. Feel free to update and I'll merge and cut a new release.

**Comment by CodySchaaf** [comment]
No worries, thanks for taking a look. Updated with suggestions.

**Comment by CodySchaaf** [comment]
@kesne how's this look now?

**Review by kesne — APPROVED** [review]
